vllm/model_executor/layers/mamba/mamba_mixer.py

```diff
@@ -331,10 +361,24 @@ def forward_cuda(self, hidden_states: torch.Tensor, output: torch.Tensor):
                 cache_indices=state_indices_tensor_p,
                 has_initial_state=has_initial_states_p,
                 query_start_loc=query_start_loc_p,
+                block_size=mamba_block_size,
+                block_idx_first_scheduled_token=block_idx_first_scheduled_token_p,
+                block_idx_last_scheduled_token=block_idx_last_scheduled_token_p,
+                initial_state_idx=block_idx_last_computed_token_p,
             )
             ssm_outputs.append(scan_out_p)
 
         if has_decode:
+            if prefix_caching_enabled:
+                state_indices_tensor_d_input = state_indices_tensor_d.gather(
+                    1, block_idx_last_computed_token_d.unsqueeze(1)
+                ).squeeze(1)
+                state_indices_tensor_d_output = state_indices_tensor_d.gather(
+                    1, block_idx_last_scheduled_token_d.unsqueeze(1)
+                ).squeeze(1)
```

Comment:
Any performance concerns about doing these gathers? I know we do similar for mamba2. Perhaps with full CUDA graph it doesn't really matter.

Reply:
TBD really. There is a PR up to fuse it into the kernel but waiting to see the perf improvement. If we see it is significant, we could revisit this.